src/main/java/org/codeforamerica/messaging/providers/twilio/TwilioCallbackController.java

```diff
@@ -49,7 +54,32 @@ private void enqueueStatusUpdate(HttpServletRequest request, String providerMess
         MessageStatus newSmsStatus = mapTwilioStatusToMessageStatus(rawMessageStatus);
         String fromPhone = request.getParameter("From");
         Map<String, String> providerError = hadError(newSmsStatus) ? buildProviderError(request) : null;
-        JobId id = jobRequestScheduler.enqueue(new SmsMessageStatusUpdateJobRequest(providerMessageId, rawMessageStatus, newSmsStatus, fromPhone, providerError));
+        jobRequestScheduler.enqueue(
+                new SmsMessageStatusUpdateJobRequest(providerMessageId, rawMessageStatus, newSmsStatus,
+                        PhoneNumber.valueOf(fromPhone),
+                        providerError));
+    }
+
+    @PostMapping(path = "/inbound", consumes = {MediaType.APPLICATION_FORM_URLENCODED_VALUE})
+    public ResponseEntity<Object> handleInbound(HttpServletRequest request) {
+        String providerMessageId = request.getParameter("MessageSid");
+        if (!twilioSignatureVerificationService.verifySignature(request)) {
+            log.error("Signature verification failed. Provider message id: {}", providerMessageId);
+            return new ResponseEntity<>("Unauthorized", HttpStatus.UNAUTHORIZED);
+        }
+        String keyword = request.getParameter("OptOutType");
+        PhoneNumber fromPhone = PhoneNumber.valueOf(request.getParameter("From"));
+        if (keyword != null) {
+            switch (keyword) {
```

Comment:
Damn, googled this and discovered that Java 18 is the first version where switch statements can handle nulls 👎